lib/finitomata.ex

```diff
@@ -40,6 +40,17 @@ defmodule Finitomata do
       type: :string,
       doc: "The FSM declaration with the syntax defined by `syntax` option."
     ],
+    forks: [
+      required: false,
+      # [AM] Allow runtime fork amending (:string)
+      # type: {:list, {:tuple, [:atom, {:or, [:atom, :string, {:list, {:or, [:atom, :string]}}]}]}},
+      type:
+        {:list,
+         {:tuple, [:atom, {:or, [{:tuple, [:atom, :atom]}, {:list, {:tuple, [:atom, :atom]}}]}]}},
```

Comment:
```
      type:
        {:list,
          {:tuple,
            [
              :atom, 
              {:or, [
                {:tuple, [:atom, :atom]},
                {:list, {:tuple, [:atom, :atom]}}
              ]}
            ]
          }
        },
```

Tried reading this spec multiple times, and failed, sorry.
Tried formating this a bit to make it easier, but failed
So it is a list of tuples, which either contain a single atom or a list of two atom tuples or list two atom tuples?

Reply:
When I started https://github.com/am-kantox/nimble_options_ex I even wanted to include a `:typed_keyword_list`, `:property`, and `:list_or_value` types which would have the syntax
```elixir
  type: {:typed_keyword_list, value_type}
  type: {:property, value_type}
  type: {:list_or_value, value_type}
```
which would have resulted in the following code here:
```elixir
  type: {:typed_keyword_list, {:list_or_value, {:tuple, [:atom, :atom]}}}
  # or even   
  type: {:typed_keyword_list, {:list_or_value, :property}}
```
I don’t recall what prevented me from doing that.

Basically, it’s a keyword (`:list, {:tuple, [:atom, _]}`) of either `state: flow_impl` or many of those like `state1: impl1, state1: impl2, state2: impl3`.

Reply:
FWIW: https://github.com/am-kantox/nimble_options_ex/issues/5